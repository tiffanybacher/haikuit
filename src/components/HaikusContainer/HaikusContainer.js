import React from 'react';

function HaikusContainer() {
  return(
    <section className="HaikusContainer">
      <p>HAIKUS CONTAINER</p>
    </section>
  );
}

export default HaikusContainer;