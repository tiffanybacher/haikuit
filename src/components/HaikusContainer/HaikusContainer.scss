.HaikusContainer {
  h2 {
    margin: 20px 0 0;
  }
}

.haikus-header {
  background: white;
  margin: 0 0 20px 0;
  padding: 20px;
}

@media only screen and (min-width: 768px) {
  .haikus-header {
    font-size: 1.1em;
    padding: 30px 50px 20px;
  }
}

@media only screen and (min-width: 1000px) {
  .haikus-header {
    font-size: 1.2em;
    padding: 40px 75px 20px;
  }
}

@media only screen and (min-width: 1200px) {
  .haikus-header {
    font-size: 1.3em;
    padding: 50px 100px 20px;
  }
}