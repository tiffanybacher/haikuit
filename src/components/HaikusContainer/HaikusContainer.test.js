import React from 'react';
import { shallow } from 'enzyme';
import { HaikusContainer, mapStateToProps } from './HaikusContainer';

describe('HaikusContainer', () => {
  let haikus;
  let history;
  let wrapper;

  beforeEach(() => {
    haikus = [{ title: 'Title', id: 111 }];
    history = { goBack: jest.fn() };
    wrapper = shallow(
      <HaikusContainer 
        haikus={haikus}
        history={history} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should go back in history', () =>{
    wrapper.instance().goBack();

    expect(history.goBack).toHaveBeenCalled();
  });

  it('should match snapshot if no haikus exist', () => {
    haikus = [];

    expect(wrapper).toMatchSnapshot();
  });

  describe('map state to props', () => {
    it('should map haikus to props', () => {
      const mockState = { 
        haikus: [{ title: 'title', id: 111 }] 
      }
      const result = mapStateToProps(mockState);
      const expected = { haikus: mockState.haikus };

      expect(result).toEqual(expected);
    });
  });
});