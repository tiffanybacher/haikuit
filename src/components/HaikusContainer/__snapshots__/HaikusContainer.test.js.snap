// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HaikusContainer should match snapshot 1`] = `
<section
  className="HaikusContainer"
>
  <div
    className="haikus-header"
  >
    <BackBtn
      goBack={[Function]}
    />
    <h2>
      YOUR HAIKUS
    </h2>
  </div>
  <Haiku
    id={111}
    key="111"
    title="Title"
  />
</section>
`;

exports[`HaikusContainer should match snapshot if no haikus exist 1`] = `
<section
  className="HaikusContainer"
>
  <div
    className="haikus-header"
  >
    <BackBtn
      goBack={[Function]}
    />
    <h2>
      YOUR HAIKUS
    </h2>
  </div>
  <Haiku
    id={111}
    key="111"
    title="Title"
  />
</section>
`;
